### DISCUSSION TITLE
verbose mode for nonlinear solve?

### DISCUSSION THREAD

**Discussion by walkersw** [firedrakeproject/firedrake#2586] [Firedrake support] [answered]
When solving a nonlinear variational problem, is there a "verbose" option?  For instance, to see how many Newton iterations it takes?  It would be useful to have for debugging.  When I use "solve(F == 0)" now, it is silent.

**Comment by pefarrell** [comment, marked as answer]
You can do `solve(F == 0, u, bcs, solver_parameters={"snes_monitor": None})` to get PETSc to tell you what's happening. (I wonder if this should be the default.)

**Comment by walkersw** [comment]
Thanks for the super-fast response.  Seems weird that *None* tells it to output info, but whatever.

**Reply by pefarrell** [reply]
I agree it's slightly weird. It's because PETSc has options that take arguments (like `-ksp_type cg`) and options that don't (like `-ksp_monitor_true_residual` or `-snes_monitor`). The former get translated into `{"ksp_type": "cg"}` and the latter into `{"snes_monitor": None}`.
